src/index.ts

```diff
@@ -19,15 +19,29 @@ export default class IntegrationTestEnvironment extends NodeEnvironment {
 
         try {
             await this.buildServicesFromConfig();
+
         } catch (e) {
             console.error("Building services failed", e.toString())
             process.exit(1)
         }
+
+        try {
+            // @ts-ignore
+            // in case the teardown failed ensure state store is cleaned out
+            await this.cleanStateStoreIfPrivate();
+        } catch (e) {
+            console.info("Error on cleaning state store", e.toString())
+            // this may not be fatal, do not die at this point
+        }
     }
 
     async teardown() {
         console.log("Tearing down integration test")
 
+        // @ts-ignore
+        // clean out our state store when we have completed tests
+        await this.cleanStateStoreIfPrivate();
```

Comment:
do we need to do it both at the beginning and the end?  Seems like just doing it at the beginning should be sufficient.  Might even be valuable to leave the state store data around after a run in case something fails and we want to investigate what the state was in S3 when the test failed

Reply:
ok sure - my thought was that cleanup is the "correct" thing to do, but sometimes it fails bc of interruption.  can leave them around.